src/libsyntax/feature_gate.rs

```diff
@@ -661,57 +665,58 @@ const EXPLAIN_STMT_ATTR_SYNTAX: &'static str =
     "attributes on non-item statements and expressions are experimental.";
 
 pub fn check_for_box_syntax(f: Option<&Features>, diag: &Handler, span: Span) {
-    if let Some(&Features { allow_box: true, .. }) = f {
+    if let Some(&Features { box_syntax: true, .. }) = f {
         return;
     }
     emit_feature_err(diag, "box_syntax", span, GateIssue::Language, EXPLAIN_BOX_SYNTAX);
 }
```

Comment:
I think that there's a number of `check_*` functions in that module which are dead code.